### DISCUSSION TITLE
Programming Help Pulse Check 🔥 [Programming Help Check-in]

### DISCUSSION THREAD

**Discussion by ebndev** [community/community#127578] [Programming Help]
![Header for the Programming Help category shows a light pink solid background with dark blue text reading “Programming Help May 2024 Check-in”. Underneath is an outline of the headings in this discussion: Community Highlights, Programming Topics, and Fostering Engagement. Next is a dark blue banner with white text that reads “Thank you for all of your contributions this month!”, along with the contributions: “238+ total discussions" and "36+ total answers”. On the left side, there is the GitHub logo in the corner. Next to it is a blue Octocat holding a laptop,  wearing a light blue cap, a white and blue striped sweater, blue jeans, and boots.](https://github.com/community/community/assets/164093511/68d0dfd8-e179-4fcf-805d-6fae2d32969f)

Hello Community! 👋 May was a busy month for the Programming Help category and we appreciate you all for taking the time to share your questions and provide answers. This is the first check-in for this category so please let us know if there's anything you'd like to see included in the future or if you have thoughts on the category overall.

_ _ _
### ✨ Community Highlights
Let's give a big round of applause to @davevad93 and @CopperEagle for all that they've contributed to the Community! You can tell they take the time to consider the questions, provide thoughtful responses, and improve the quality of GitHub Discussions.

Here are examples of them commenting on discussions with code blocks, explaining steps for implementation, and linking to other resources. It has been so helpful for other users to receive such detailed responses, so thank you both for sharing your expertise and passion as developers. 💐

- @CopperEagle commenting on @judahpaul16's discussion about [NLP Models for Action Routing](https://github.com/orgs/community/discussions/125994).

- @davevad93 answering @daniellelamb's question on [CSS Borders and Alignment](https://github.com/orgs/community/discussions/126285).

Thank you to all users who provided answers this month! 💐💐💐 We hope you'll continue to engage and contribute, as Programming Help spans across so many different topics and technologies.

@And2224 @arashtabaa @dev188007 @fdistasio @GM7Avila @judahpaul16 @liker912 @LiteBrite82 @Mahmud-cse @MarcosApodaca @nikhil-shuklaa @pkprajapati7402 @quietpulse @Sackitey68 @sajadh76 @vencomweb @Vikernes27666 @ViniciusMarins

_ _ _
### 💻 Programming Topics
With the Programming Help category getting such a vast range of questions, here's a quick pulse check on popular technologies and where the Community could use more support.

🟢 The following topics seem to have consistent engagement and are more likely to have responses or answers.

|            | Technology |
| --------   | :--------- |
| Languages  | Assembly, HTML & CSS, JavaScript, PHP, Python |
| Frameworks | Astro, Fabric.js |
| Other      | Data Modeling, VSCode, Web-Development  |

🟡 Below are frequently asked questions around topics that the Community could use more support in. If you know of any developers or other online communities interested in these areas, please invite them to participate in Discussions. 😁

|            | Technology |
| --------   | :--------- |
| Languages  | Java       |
| Frameworks | Flutter, Next.js, React |
| Hardware   | Arduino, Raspberry Pi   |
| Other      | Cloud                   |

_ _ _
### 🚀 Fostering Engagement
Answering questions isn't the only way to engage in the category! Whether posting a discussion or commenting, there are things you can do that may encourage more participation.

As a user posting a question:
- Give your post a good title, telling folks at first glance what your post is about.
  - Discussions titled with "Please help me" can be viewed as spammy.
- Provide context for your problem: what you're working on, what errors you've encountered, and what you've done to debug.
  - Often, posting a wall of code or screenshot of your terminal doesn't make it clear what you need help with.
- Please follow up. 💖 This helps folks who have the same question in the future know where they can go!
  - If someone solves your problem, you can mark their response as the answer. 
  - If you've solved the problem on your own, you can add a comment or close the discussion.
 
Here’s a great example of a [discussion](https://github.com/orgs/community/discussions/122182) where @RotemTon has a clear title, briefly describes the issue, what they’ve tried and why it didn’t work, followed by the code in a code block. Although they were not able to fully solve their problem through Discussions, they later followed up with their solution for others who may run into the issue in the future and thanked @markolalovic, who gave them suggestions.

For users commenting on posts:
- Ask for more context.
  - Even if you are unable to solve the problem, guiding OP with questions might help them debug and clarify what their problem is.
- Link to resources.
  - Love a YouTube tutorial or stumbled across an article talking about OP's topic? Feel free to share what you think might be helpful.
- Direct them to past discussions.
  - If there are duplicate posts in the Community, you can link OP to previous discussions and encourage them to comment or find a solution there. Folks who have engaged in the original discussion will get notified of activity in the post and could possibly answer new questions.
 
As you all continue to learn and grow in your coding journeys, thanks for being a part of the GitHub Community!

**Comment by davevad93** [comment]
Thank you very much for the mention @ebndev, actually I've been a GitHub user for a few years but I discovered the Community Discussions only a few months ago (my bad XD). I really like this space and will continue to contribute by helping other users whenever I can.

**Reply by ebndev** [reply]
haha no worries, it can be a bit hidden but we're happy you're here now!

**Comment by CopperEagle** [comment]
Thank you so much for your kind words @ebndev :heart: Community Discussions are really fun and you learn a lot either way!

Speaking of learning though, unfortunately, I will need to reduce my activity a bit in the next few months ;/... Exam session is looking at me like it's lunch time :sweat_smile: 

Thanks again and have a nice weekend!

**Reply by davevad93** [reply]
Good luck with your exam prep @CopperEagle, stay focused! 💪

**Reply by CopperEagle** [reply]
Thank you @davevad93, you rock :guitar: :+1:

**Reply by ebndev** [reply]
Ditto what davevad93 said, good luck on exams! We completely understand capacity ebbs and flows- the Community will still be here if you decide to return 😄

**Reply by pkprajapati7402** [reply]
Thank dude

**Comment by liker912** [comment]
Thank you so much @ebndev for your mention. I also discovered Community Discussions just a few months ago, and it's really cool that there's a place where you can get help and receive help. I will try to maintain my level of activity.

**Reply by ebndev** [reply]
Awesome! Just curious: what tech stacks do you typically use?

**Reply by liker912** [reply]
As a Fullstack developer, I often use PHP and Node.js for backend development, and less frequently Golang or Java. For frontend development, I primarily use React/Next.js, and less often Vue or Angular.
